tests/test_endpoint_correction.py

```diff
@@ -0,0 +1,226 @@
+from jax.config import config; config.update("jax_enable_x64", True)
+import jax
+
+import copy
+import functools
+
+from rdkit import Chem
+from fe import endpoint_correction
+from timemachine.integrator import langevin_coefficients
+from timemachine.potentials import bonded
+from timemachine import constants
+import numpy as np
+
+from pathlib import Path
+from ff.handlers.deserialize import deserialize_handlers
+from ff import Forcefield
+
+
+def get_romol_conf(mol):
+    """Coordinates of mol's 0th conformer, in nanometers"""
+    conformer = mol.GetConformer(0)
+    guest_conf = np.array(conformer.GetPositions(), dtype=np.float64)
+    return guest_conf/10 # from angstroms to nm
+
+def setup_system():
+
+    root = Path(__file__).parent.parent
+    path_to_ligand = str(root.joinpath('tests/data/ligands_40.sdf'))
+    path_to_ff = str(root.joinpath('ff/params/smirnoff_1_1_0_ccc.py'))
+
+    with open(path_to_ff) as f:
+        ff_handlers = deserialize_handlers(f.read())
+
+    forcefield = Forcefield(ff_handlers)
+    suppl = Chem.SDMolSupplier(path_to_ligand, removeHs=False)
+    all_mols = [x for x in suppl]
+    mol_a = copy.deepcopy(all_mols[1])
+
+
+    # test identity transformation
+    # mol_a, _, _, forcefield = relative.hif2a_ligand_pair
+    mol_b = Chem.Mol(mol_a)
+
+    # parameterize with bonds and angles
+    bond_params_a, bond_idxs_a = forcefield.hb_handle.parameterize(mol_a)
+    angle_params_a, angle_idxs_a = forcefield.ha_handle.parameterize(mol_a)
+
+    bond_params_b, bond_idxs_b = forcefield.hb_handle.parameterize(mol_b)
+    angle_params_b, angle_idxs_b = forcefield.ha_handle.parameterize(mol_b)
+
+    box = np.eye(3)*100.0
+
+    bond_fn = functools.partial(
+        bonded.harmonic_bond,
+        bond_idxs=np.concatenate([bond_idxs_a, bond_idxs_b+mol_a.GetNumAtoms()]),
+        params=np.concatenate([bond_params_a, bond_params_b]),
+        box=box,
+        lamb=None
+    )
+
+    angle_fn = functools.partial(
+        bonded.harmonic_angle,
+        angle_idxs=np.concatenate([angle_idxs_a, angle_idxs_b+mol_a.GetNumAtoms()]),
+        params=np.concatenate([angle_params_a, angle_params_b]),
+        box=box,
+        lamb=None
+    )
+
+    core_idxs = []
+    core_params = []
+
+    for a in mol_a.GetAtoms():
+        if a.IsInRing():
+            core_idxs.append((a.GetIdx(), a.GetIdx() + mol_a.GetNumAtoms()))
+            # if the force constant is too high, no k/t combination can generate
```

Comment:
If we set the harmonic core restraints to too high of a value, then there's seems to be no suitable choice of k_translation/k_rotation that can maximize the overlap.